### PR TITLE
Modernize Microsoft.Scripting/Runtime files

### PR DESCRIPTION
**Pull request by gpetrou**

### CODE REVIEW

File: Src/Microsoft.Scripting/Runtime/ScriptDomainManager.cs [hunk 1]
```diff
@@ -42,17 +42,11 @@ public sealed class ScriptDomainManager {
...
 
-        public SharedIO SharedIO {
-            get { return _sharedIO; }
-        }
+        public SharedIO SharedIO { get; }
 
-        public DynamicRuntimeHostingProvider Host {
-            get { return _hostingProvider; }
-        }
+        public DynamicRuntimeHostingProvider Host => _hostingProvider;
```
Comment (slozier) [inline Src/Microsoft.Scripting/Runtime/ScriptDomainManager.cs]: Use auto property instead?

File: Src/Microsoft.Scripting/Runtime/ScriptDomainManager.cs [hunk 2]
```diff
@@ -42,17 +42,11 @@ public sealed class ScriptDomainManager {
...
 
-        public DlrConfiguration Configuration {
-            get { return _configuration; }
-        }
+        public DlrConfiguration Configuration => _configuration;
```
Comment (slozier) [inline Src/Microsoft.Scripting/Runtime/ScriptDomainManager.cs]: auto property?

File: Src/Microsoft.Scripting/Runtime/TokenInfo.cs [hunk 3]
```diff
@@ -25,18 +25,18 @@ public struct TokenInfo : IEquatable<TokenInfo> {
         private SourceSpan _span;
         
         public TokenCategory Category {
```
Comment (slozier) [inline Src/Microsoft.Scripting/Runtime/TokenInfo.cs]: Why not make all of these auto properties?

File: Src/Microsoft.Scripting/Runtime/ContextId.cs [hunk 4]
```diff
@@ -57,21 +55,16 @@ public struct ContextId : IEquatable<ContextId> {
...
-            return ContextId.Empty;
+            return Empty;
         }
 
-        public int Id {
-            get {
-                return _id;
-            }
-        }
+        public int Id => _id;
```
Comment (slozier) [inline Src/Microsoft.Scripting/Runtime/ContextId.cs]: Make it an auto property?

File: Src/Microsoft.Scripting/Runtime/DlrConfiguration.cs [hunk 5]
```diff
@@ -30,18 +31,12 @@ internal sealed class LanguageConfiguration {
...
         
-        public LanguageContext LanguageContext {
-            get { return _context; }
-        }
+        public LanguageContext LanguageContext => _context;
 
-        public AssemblyQualifiedTypeName ProviderName {
-            get { return _providerName; }
-        }
+        public AssemblyQualifiedTypeName ProviderName => _providerName;
```
Comment (slozier) [inline Src/Microsoft.Scripting/Runtime/DlrConfiguration.cs]: These could be auto properties...

File: Src/Microsoft.Scripting/Runtime/NotNullAttribute.cs [hunk 6]
```diff
@@ -32,7 +32,5 @@ public sealed class NotNullAttribute : Attribute {
     /// </summary>
     [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false, Inherited = false)]
     public sealed class NotNullItemsAttribute : Attribute {
-        public NotNullItemsAttribute() {
```
Comment (slozier) [inline Src/Microsoft.Scripting/Runtime/NotNullAttribute.cs]: Although unlikely, if you remove the this constructor and someone comes along and add a constructor overload (e.g. with an argument) then the default constructor will be gone.

File: Src/Microsoft.Scripting/Runtime/ScriptCode.cs [hunk 7]
```diff
@@ -31,13 +31,9 @@ public abstract class ScriptCode {
...
 
-        public LanguageContext LanguageContext {
-            get { return _sourceUnit.LanguageContext; }
-        }
+        public LanguageContext LanguageContext => _sourceUnit.LanguageContext;
 
-        public SourceUnit SourceUnit {
-            get { return _sourceUnit; }
-        }
+        public SourceUnit SourceUnit => _sourceUnit;
```
Comment (slozier) [inline Src/Microsoft.Scripting/Runtime/ScriptCode.cs]: auto property?

File: Src/Microsoft.Scripting/Runtime/NotNullAttribute.cs (hunk 6 above)
Comment (gpetrou) [reply Src/Microsoft.Scripting/Runtime/NotNullAttribute.cs]: I think we had removed the NotNullAttribute constructor in the same file sometime ago as well, so this makes the changes consistent.

**Pull request merged**